dask/base.py

```diff
@@ -416,14 +416,32 @@ def tokenize(*args, **kwargs):
     return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
 
 
+def dont_optimize(dsk, keys):
+    return dsk
+
+
+def optimization_function(obj):
+    if isinstance(obj, type):
+        cls = obj
+    else:
+        cls = type(obj)
+    name = cls.__name__.lower() + '_optimize' # dask.set_options(array_optimize=foo)
```

Comment:
I agree that this would solve the explicitness problem without having the often-repetitive class attribute.  However it also puts the same potentially error prone code in many places.  I suppose we could find a nice way to organize things though.

Thoughts on checking `cls._optimization_name` and using that if it exists or, if not, defaulting to `cls.__name__.lower()`.  This allows collections to be explicit if they prefer but also allows the common case to be easy.  It also keeps the logic here centralized in `base.py`.